### PR TITLE
Remove MLton tools, documentation, unsupported libraries, etc.

### PR DESCRIPTION
**Pull request by shwestrick**
@MatthewFluet wanted to get your opinion on this. These changes seem nice to keep things simple, avoid any confusion between `mpl` and `mlton`, and make builds a teeny bit faster (don't have to compile the tools or type-check the additional libraries). I figure we can always pull stuff over from `mlton` in the future if needed. Do you anticipate any problems?

Here's a summary of what I removed:
  - mlyacc and mllex. Since mpl requires mlton or sml/nj to build from source, these tools will be available elsewhere.
  - mlprof. Currently unsupported with mpl, and likely won't be supported in the near future.
  - cm2mlb and mlnlffigen, which are provided by mlton.
  - man pages for mlton and removed tools.
  - libraries: mlrisc-lib, mlnlffi-lib, ckit-lib. These are available through mlton if needed.
  - packaging infrastructure. We can port this over for mpl in the future if needed.
  - ide stuff (emacs modes, etc) which are available through mlton if needed.
  - mlton development guides
  - mlton changelog
  - mlton website sources
  - unsupported regression tests

----------

**EDIT** (see discussion below):
  - although yacc/lex tools have been removed, the mlyacc-lib and mllpt-lib are still available, because these are used by smlnj-lib.
  - also removed cml library
  - disabled unsupported codegens (llvm, x86, arm64). Only C codegen is available.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -113,15 +114,9 @@ endef
 
 .PHONY: libraries-no-check
 libraries-no-check:
-	$(MAKE) -C "$(SRC)/lib/ckit-lib"
-	$(call LIBRARIES_NO_CHECK_TEMPLATE,ckit-lib,/ckit/src,)
 	$(call LIBRARIES_NO_CHECK_TEMPLATE,cml,,'!' -path '*/tests/*')
```
Comment (MatthewFluet) [inline Makefile:117]: Why keep the CML library?  I don't think it works with MaPLe?

File: Makefile [hunk 2]
```diff
@@ -113,15 +114,9 @@ endef
...
 	$(MAKE) -C "$(SRC)/lib/mllpt-lib"
 	$(call LIBRARIES_NO_CHECK_TEMPLATE,mllpt-lib,/ml-lpt/lib,)
```
Comment (MatthewFluet) [inline Makefile:119]: Why keep the ML-LPT library?

File: Makefile [hunk 3]
```diff
@@ -113,15 +114,9 @@ endef
...
-	$(MAKE) -C "$(SRC)/lib/mlnlffi-lib"
-	$(call LIBRARIES_NO_CHECK_TEMPLATE,mlnlffi-lib,,)
-	$(MAKE) -C "$(SRC)/lib/mlrisc-lib"
-	$(call LIBRARIES_NO_CHECK_TEMPLATE,mlrisc-lib,/MLRISC,'!' -path '*/demo/*' '!' -path '*/Tools/*' '!' -path './autoload.sml' '!' -path './make*.sml')
 	$(call LIBRARIES_NO_CHECK_TEMPLATE,mlyacc-lib,,)
 	$(MAKE) -C "$(SRC)/lib/smlnj-lib"
 	$(call LIBRARIES_NO_CHECK_TEMPLATE,smlnj-lib,/smlnj-lib,'!' -path '*/examples/*' '!' -path '*/tests/*' '!' -path '*/Tests/*')
```
Comment (MatthewFluet) [inline Makefile:122]: Why keep the SML/NJ library?

File: Makefile [hunk 4]
```diff
@@ -133,12 +128,7 @@ endef
 
 .PHONY: libraries-check
 libraries-check:
-	$(call LIBRARIES_CHECK_TEMPLATE,ckit-lib)
 	$(call LIBRARIES_CHECK_TEMPLATE,cml)
-	$(call LIBRARIES_CHECK_TEMPLATE,mllpt-lib)
-	$(call LIBRARIES_CHECK_TEMPLATE,mlnlffi-lib)
-	$(call LIBRARIES_CHECK_TEMPLATE,mlrisc-lib)
-	$(call LIBRARIES_CHECK_TEMPLATE,mlyacc-lib)
```
Comment (MatthewFluet) [inline Makefile:141]: Why omit checking the MLYacc Library?

File: Makefile [hunk 5]
```diff
@@ -133,12 +128,7 @@ endef
 
 .PHONY: libraries-check
 libraries-check:
-	$(call LIBRARIES_CHECK_TEMPLATE,ckit-lib)
 	$(call LIBRARIES_CHECK_TEMPLATE,cml)
-	$(call LIBRARIES_CHECK_TEMPLATE,mllpt-lib)
```
Comment (MatthewFluet) [inline Makefile:138]: If keeping the ML-LPT library, why omit check it?

**Review by MatthewFluet — COMMENTED** [review]
I wouldn't anticipate any problems, other than possibly merge conflicts when trying to sync with upstream.  But, that shouldn't be too bad, because you are just deleting files.

If you really wanted to speed up your builds, you could consider removing the native codegens, since MaPLe doesn't support them.

**Comment by shwestrick** [comment]
Thanks for the feedback! I went ahead and removed CML (0a9a9da0fff78849d705b26a6c3bec21294baa08) and put the checking back for mllpt and mlyacc (fbff6c75fee991a3d17cf5f31fdc3f52fcfbe8b1).

Basically, I want to keep the SML/NJ library around because we have *occasionally* used it... and then SML/NJ depends on lpt and yacc libraries:
  - `smlnj-lib/smlnj-lib/HTML4/html4-lib.mlb` uses `$(SML_LIB)/mllpt-lib/mllpt-lib.mlb`
  - `smlnj-lib/smlnj-lib/HTML/html-lib.mlb` uses `$(SML_LIB)/mlyacc-lib/mlyacc-lib.mlb`

So, it seems simplest to just keep them all around for now.

**Comment by shwestrick** [comment]
> If you really wanted to speed up your builds, you could consider removing the native codegens, since MaPLe doesn't support them.

Good suggestion! I've been meaning to do that for a while. And now, done (bc1034bcb562254fb630ec7cef084454cce8a25f).

**Pull request merged**